Automattic/jetpack
# Masterbar: Copy module code to package

This PR copies the Masterbar module codebase to the `masterbar` package. In the process:
- Methods that only live in the Jetpack plugin have been replaced by corresponding methods available in packages. For  example: `Jetpack::is_module_active` has been replaced by `( new Modules() )->is_module_active`
- The `JETPACK__VERSION` used for cache busting when enqueueing scripts and styles is now replaced by `Main::PACKAGE_VERSION`
- Translations text domain is updated to `jetpack-masterbar` from `jetpack`
- Updated the build process so that `rtl` files are now generated with a [custom config](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/tools/webpack.config.css.js#L179) (we used to refer to them as weird RTL entries because `-rtl` was appended instead of `.rtl`) and the corresponding code references as well.
- Updated all scripts and css files to be loaded from the `dist` folder
- Started enqueuing all package scripts and styles via the `Assets` package
- Fixed close to 270 phan issues
- Added more unit tests, related mostly to enqueuing assets

### Other information:


## Jetpack product discussion


pfwV0U-3U-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Code Review since the package is not used anywhere yet.
* Migrated unit tests should all pass
* Try running `jetpack build packages/masterbar` with/without `--production` and inspect the generated assets in the `dist` folder
